#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/CircleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/appearances/protos/Parquetry.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/robots/universal_robots/protos/UR10e.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/devices/robotiq/protos/Robotiq3fGripper.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/solids/protos/SolidBox.proto"

WorldInfo {
  basicTimeStep 5
}
Viewpoint {
  orientation 0.22031886599877806 0.7783165325896566 -0.5879481034773615 0.8973921838601566
  position -3.9845921108062967 2.352883493101669 4.086707679773413
}
TexturedBackground {
}
TexturedBackgroundLight {
}
CircleArena {
  radius 1.42
  floorAppearance Parquetry {
    type "light strip"
  }
}
CircleArena {
  translation 0 0 -0.01
  name "inner area"
  radius 0.2
  floorAppearance Parquetry {
    type "light strip"
  }
  wallHeight 0.05
}
DEF green_box SolidBox {
  translation -0.5554612606558823 -0.3841082633065622 0.06
  name "green_box"
  size 0.1 0.1 0.1
  appearance PBRAppearance {
    baseColor 0 1 0
    roughness 0.5
    metalness 0.5
  }
  physics Physics {
    density -1
    mass 0.1
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      0.0016666666666666674 0.0016666666666666674 0.0016666666666666674
      0 0 0
    ]
  }
}
DEF red_box SolidBox {
  translation -0.16816633312970586 0.6185339503490799 0.06
  name "red_box"
  size 0.1 0.1 0.1
  appearance PBRAppearance {
    baseColor 1 0 0
    roughness 0.5
    metalness 0.5
  }
  physics Physics {
    density -1
    mass 0.1
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      0.0016666666666666674 0.0016666666666666674 0.0016666666666666674
      0 0 0
    ]
  }
}
DEF blue_box SolidBox {
  translation -0.1177503292235289 -0.6596783426292468 0.06
  name "blue_box"
  size 0.1 0.1 0.1
  appearance PBRAppearance {
    baseColor 0 0 1
    roughness 0.5
    metalness 0.5
  }
  physics Physics {
    density -1
    mass 0.1
    centerOfMass [
      0 0 0
    ]
    inertiaMatrix [
      0.0016666666666666674 0.0016666666666666674 0.0016666666666666674
      0 0 0
    ]
  }
}
DEF UR10e UR10e {
  hidden rotation_12 0 0 -1 0.5200000013262689
  hidden rotation_16 0 0 -1 0.5200000008057115
  controller "ur3e_with_gripper"
  supervisor TRUE
  selfCollision FALSE
  toolSlot [
    Camera {
      rotation 0 0 1 1.5708
      name "tip_camera"
      width 128
      height 128
    }
    GPS {
      translation 0 0.21 0
      name "gps_tip"
    }
    Robotiq3fGripper {
    }
  ]
}
